File: 2025/day02/two.rb
require 'debug'

data = File.readlines(
  File.expand_path("../../../data/2025/day02/#{ARGV[-1]}", __FILE__),
  sep=',',
)
  .map! { |s| Range.new(*s.split('-').map!(&:to_i)) }

def count_jigajolts(ranges)
  sum = 0
  ranges.each do |range|
    range.each do |number|
      s = number.to_s
      len = s.length
      (1..len/2).each do |width|
        if len % width != 0
          next
        end
        if s[0, width] * (len / width) == s
          sum += number
          break
        end
      end
    end
  end
  return sum
end


puts count_jigajolts(data)
# 33986149340
# 4174379265

File: 2025/day02/two.py
import sys
from os.path import dirname, join


FILE = join(dirname(__file__), '../../data/2025/day02/', sys.argv[-1])
with open(FILE) as f:
    data = [line.split('-') for line in f.read().strip().split(',')]


def count_naive(data):
    _sum = 0
    for start, end in data:
        r = range(int(start), int(end)+1)
        for e in r:
            s = str(e)
            cur_len = len(s)
            for width in range(1, cur_len//2 + 1):
                if cur_len % width != 0:
                    continue
                if s[:width] * (cur_len // width) == s:
                    _sum += e
                    break

    return _sum


# 69564213293
print(count_naive(data))

# from timeit import repeat
# print(repeat(
#     'count_naive(data)',
#     setup="from __main__ import count_naive",
#     globals=globals(),
#     repeat=3,
#     number=1,
# ))

